presto-main/src/main/java/com/facebook/presto/server/RequestErrorTracker.java

```diff
@@ -44,38 +47,64 @@
 import static com.facebook.presto.util.Failures.WORKER_NODE_ERROR;
 import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
 import static java.util.concurrent.TimeUnit.NANOSECONDS;
 import static java.util.concurrent.TimeUnit.SECONDS;
 
 @ThreadSafe
 public class RequestErrorTracker
 {
     private static final Logger log = Logger.get(RequestErrorTracker.class);
-
+    private static final List<Duration> LEAF_BACKOFF_DELAY_INTERVALS = ImmutableList.<Duration>builder()
+            .add(new Duration(0, MILLISECONDS))
+            .add(new Duration(10, MILLISECONDS))
+            .add(new Duration(20, MILLISECONDS))
+            .add(new Duration(30, MILLISECONDS))
+            .add(new Duration(40, MILLISECONDS))
+            .add(new Duration(50, MILLISECONDS))
+            .add(new Duration(60, MILLISECONDS))
+            .add(new Duration(100, MILLISECONDS))
+            .add(new Duration(500, MILLISECONDS))
+            .build();
     private final Object id;
     private final URI uri;
     private ErrorCodeSupplier errorCode;
     private String nodeErrorMessage;
     private final ScheduledExecutorService scheduledExecutor;
     private final String jobDescription;
     private final Backoff backoff;
-
+    public boolean isLeaf;
     private final Queue<Throwable> errorsSinceLastSuccess = new ConcurrentLinkedQueue<>();
 
     public RequestErrorTracker(Object id, URI uri, ErrorCodeSupplier errorCode, String nodeErrorMessage, Duration maxErrorDuration, ScheduledExecutorService scheduledExecutor, String jobDescription)
+    {
+        this(id, uri, errorCode, nodeErrorMessage, maxErrorDuration, scheduledExecutor, jobDescription, false);
+    }
+
+    private RequestErrorTracker(Object id, URI uri, ErrorCodeSupplier errorCode, String nodeErrorMessage, Duration maxErrorDuration, ScheduledExecutorService scheduledExecutor, String jobDescription, boolean isLeaf)
     {
         this.id = requireNonNull(id, "id is null");
         this.uri = requireNonNull(uri, "uri is null");
         this.errorCode = requireNonNull(errorCode, "errorCode is null");
         this.nodeErrorMessage = requireNonNull(nodeErrorMessage, "nodeErrorMessage is null");
         this.scheduledExecutor = requireNonNull(scheduledExecutor, "scheduledExecutor is null");
-        this.backoff = new Backoff(requireNonNull(maxErrorDuration, "maxErrorDuration is null"));
+        this.backoff = initBackoff(maxErrorDuration, isLeaf);
         this.jobDescription = requireNonNull(jobDescription, "jobDescription is null");
+        this.isLeaf = isLeaf;
+    }
+
+    private Backoff initBackoff(Duration maxErrorDuration, boolean isLeaf)
+    {
+        if (!isLeaf) {
```

Comment:
So all leaves get this special backoff even if they are not in graceful shutdown mode ?